samples/ingestion/ingestion-client/Connector/BatchClient.cs

```diff
@@ -118,6 +119,13 @@ private static async Task<HttpResponseMessage> SendHttpRequestMessage(HttpReques
 
                 return responseMessage;
             }
+            catch (HttpRequestException requestException)
+                when (requestException.InnerException != null &&
+                requestException.InnerException is SocketException socketException &&
+                socketException.SocketErrorCode == SocketError.TimedOut)
```

Comment:
Ideally retry on any socker error, right? You can't really call it a TimeoutException anymore but I do think we should handle this broader category of failures and attempt retry

Reply:
I don't think we should retry in all other cases as most of them are not transient, so no need to retry there. 
I think I'd prefer to only check for the timeout for now, and add other cases if needed later.